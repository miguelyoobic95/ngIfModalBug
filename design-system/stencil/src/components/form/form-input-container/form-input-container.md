---
name: Container
category: Form Components
---

The container should be used to wrap a form component.

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`label`|string|   |the input label|
|`description`|string|   |the form input description|
|`required`|boolean|`false` |`true` adds required red star to label|
|`hint`|string|   |adds hint text under the form input|

```yoo-form-input-container.html
        <yoo-form-input-container label="my label" description="A form container with a form input as content." hint="this is a hint" required="true">
            <div >
                <yoo-form-input></yoo-form-input>
            </div>
        </yoo-form-input-container>
        <br><br>
         <yoo-form-input-container label="this is a form container with a checkbox">
            <div >
                <yoo-form-checkbox text="this is a checkbox" class="accent"></yoo-form-checkbox>
            </div>
        </yoo-form-input-container>
        <br><br>
         <yoo-form-input-container>
            <div >
                 <yoo-form-input label="my label" placeholder="my place holder" description="A form container with a form input as content." placeholdertolabel="true"></yoo-form-input> 
            </div>
        </yoo-form-input-container>
```