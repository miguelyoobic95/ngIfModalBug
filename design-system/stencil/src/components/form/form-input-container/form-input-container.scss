 @mixin text-color($color, $color-20) {
     :host(.#{$color}) {
         .hint {
             color: cssvar($color-20);
         }
         .description {
             color: cssvar($color);
         }
     }
 }

 :host {
     display: block; //
     margin-bottom: 1rem;
     .label {
         font-size: 0.875rem;
         font-weight: normal;
         line-height: 1.5;
         letter-spacing: 2px;
         text-align: left;
         color: cssvar(text-color);
         margin-bottom: 0.25rem;
         font-variant-caps: all-petite-caps;
         .label-required {
             margin-left: 0.25rem;
             color: cssvar(danger);
         }
     }
     .hint {
         font-size: 0.75rem;
         line-height: 1.33;
         text-align: left;
         color: cssvar(text-color);
         margin-top: 0.25rem;
         margin-bottom: 0.25rem;
     }
     .description {
         color: cssvar(dark);
         font-size: 0.875rem;
     }
 }

 @include text-color(accent, accent-20);
 @include text-color(dark, dark-20);
 @include text-color(danger, danger-20);
 @include text-color(success, success-20);
 @include text-color(warning, warning-20);
 @include text-color(info, info-20);
 @include text-color(stable, stable);