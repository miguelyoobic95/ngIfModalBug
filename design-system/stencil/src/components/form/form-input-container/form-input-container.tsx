import { Component, Prop, Element } from '@stencil/core';
import { IFormInputContainer } from '@shared/interfaces';

@Component({
    tag: 'yoo-form-input-container',
    styleUrl: 'form-input-container.scss',
    scoped: true
})
export class YooFormInputContainerComponent implements IFormInputContainer {

    @Prop() label: string;
    @Prop() description: string;
    @Prop() hint: string;
    @Prop() required: boolean;

    @Element() host: HTMLStencilElement;

    render(): JSX.Element {
        return [
            this.description ? <div class="description" innerHTML={this.description}> </div> : '',
            this.label ?
                <div class="label">
                    <span innerHTML={this.label}></span>
                    {this.required ? <span class="label-required">*</span> : null}
                </div> : null,
            <div class="content-container">
                <slot />
            </div>,
            this.hint ? <div class="hint">{this.hint}</div> : ''
        ];
    }
}
