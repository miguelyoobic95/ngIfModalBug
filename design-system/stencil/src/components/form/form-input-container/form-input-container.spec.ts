import { TestWindow } from '@stencil/core/testing';
import { YooFormInputContainerComponent } from './form-input-container';

describe('YooFormInputContainerComponent', () => {
    it('should build', () => {
        expect(new YooFormInputContainerComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it ('Should have description content', async () => {
            let window = new TestWindow();
            let element = await window.load({
               components: [YooFormInputContainerComponent],
               html: '<yoo-form-input-container description="i am description"></yoo-form-input-container>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have label content', async () => {
            let window = new TestWindow();
            let element = await window.load({
               components: [YooFormInputContainerComponent],
               html: '<yoo-form-input-container label="i am label"></yoo-form-input-container>'
            });
            expect(element).toMatchSnapshot();
        });

          it ('Should have hint content', async () => {
            let window = new TestWindow();
            let element = await window.load({
               components: [YooFormInputContainerComponent],
               html: '<yoo-form-input-container hint="i am hint"></yoo-form-input-container>'
            });
            expect(element).toMatchSnapshot();
        });

         it ('Should have all attributes', async () => {
            let window = new TestWindow();
            let element = await window.load({
               components: [YooFormInputContainerComponent],
               html: '<yoo-form-input-container hint="i am hint" label="i am label" description="i am description"</yoo-form-input-container>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have all attributes and slot content', async () => {
            let window = new TestWindow();
            let element = await window.load({
               components: [YooFormInputContainerComponent],
               html: '<yoo-form-input-container hint="i am hint" label="i am label" description="i am description" <div > <yoo-form-checkbox text="this is a checkbox" class="accent"></yoo-form-checkbox> </div> </yoo-form-input-container>'
            });
            expect(element).toMatchSnapshot();
        });
    });
});
