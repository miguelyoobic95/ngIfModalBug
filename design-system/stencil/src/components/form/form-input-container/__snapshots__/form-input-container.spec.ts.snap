// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormInputContainerComponent Snapshots Should have all attributes 1`] = `
<yoo-form-input-container
  <=""
  class="hydrated"
  data-yoo-form-input-container-host=""
  description="i am description"
  hint="i am hint"
  label="i am label"
  yoo-form-input-container=""
>
  
  <div
    class="description"
    data-yoo-form-input-container=""
  >
    i am description
     
  </div>
  <div
    class="label"
    data-yoo-form-input-container=""
  >
    <span
      data-yoo-form-input-container=""
    >
      i am label
    </span>
    
  </div>
  <div
    class="content-container"
    data-yoo-form-input-container=""
    data-yoo-form-input-container-slot=""
  >
    
  </div>
  <div
    class="hint"
    data-yoo-form-input-container=""
  >
    i am hint
  </div>
</yoo-form-input-container>
`;

exports[`YooFormInputContainerComponent Snapshots Should have all attributes and slot content 1`] = `
<yoo-form-input-container
  <div=""
  class="hydrated"
  data-yoo-form-input-container-host=""
  description="i am description"
  hint="i am hint"
  label="i am label"
>
  
  
  
  
  <div
    class="description"
    data-yoo-form-input-container=""
  >
    i am description
     
  </div>
  <div
    class="label"
    data-yoo-form-input-container=""
  >
    <span
      data-yoo-form-input-container=""
    >
      i am label
    </span>
    
  </div>
  <div
    class="content-container"
    data-yoo-form-input-container=""
    data-yoo-form-input-container-slot=""
  >
    
     
    <yoo-form-checkbox
      class="accent"
      text="this is a checkbox"
    />
      
  </div>
  <div
    class="hint"
    data-yoo-form-input-container=""
  >
    i am hint
  </div>
</yoo-form-input-container>
`;

exports[`YooFormInputContainerComponent Snapshots Should have description content 1`] = `
<yoo-form-input-container
  class="hydrated"
  data-yoo-form-input-container-host=""
  description="i am description"
>
  
  <div
    class="description"
    data-yoo-form-input-container=""
  >
    i am description
     
  </div>
  
  <div
    class="content-container"
    data-yoo-form-input-container=""
    data-yoo-form-input-container-slot=""
  >
    
    
    
  </div>
  
</yoo-form-input-container>
`;

exports[`YooFormInputContainerComponent Snapshots Should have hint content 1`] = `
<yoo-form-input-container
  class="hydrated"
  data-yoo-form-input-container-host=""
  hint="i am hint"
>
  
  
  <div
    class="content-container"
    data-yoo-form-input-container=""
    data-yoo-form-input-container-slot=""
  >
    
    
  </div>
  <div
    class="hint"
    data-yoo-form-input-container=""
  >
    i am hint
  </div>
</yoo-form-input-container>
`;

exports[`YooFormInputContainerComponent Snapshots Should have label content 1`] = `
<yoo-form-input-container
  class="hydrated"
  data-yoo-form-input-container-host=""
  label="i am label"
>
  
  
  <div
    class="label"
    data-yoo-form-input-container=""
  >
    <span
      data-yoo-form-input-container=""
    >
      i am label
    </span>
    
  </div>
  <div
    class="content-container"
    data-yoo-form-input-container=""
    data-yoo-form-input-container-slot=""
  >
    
    
    
  </div>
  
</yoo-form-input-container>
`;
